src/App.js

```diff
@@ -1,19 +1,42 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
 
 const App = () => {
+  const [chatData, setChatData] = useState(chatMessages);
+
+  const updateChatData = (updatedChatData) => {
+    setChatData(
+      chatData.map((entry) =>
+        entry.id === updatedChatData.id ? updatedChatData : entry
+      )
+    );
+  };
+
+  const calcTotalHearts = (chatData) => {
+    return chatData.reduce((total, entry) => {
+      return total + entry.liked;
```

Comment:
It's a little surprising to add a boolean directly to a number. It works, since JS coerces the boolean into an int (true is 1 and false is 0), but it might be nice to use a ternary to explicitly show the conversion.